solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -99,4 +118,125 @@ impl FlashbotsApi {
         ensure!(status.is_success(), "status {}: {:?}", status, body);
         Ok(())
     }
+
+    /// Query gas_price for the current network state (simplest one for Flashbots)
+    pub async fn gas_price(&self) -> Result<EstimatedGasPrice> {
```

Comment:
I'm wondering if this should return `Result<GasPrice1559>` because based on the code below that is always the case instead of the full EstimatedGasPrice struct. But not sure if that makes code using this function harder.

Reply:
Not bad idea, however I would leave it as is, since the caller works with EstimatedGasPrice, and also, it will be easier to refactor the submitting strategies later.